gradle-typescript/src/main/java/com/gradlets/gradle/typescript/idea/CreateTsConfigTask.java

```diff
@@ -81,22 +83,14 @@ public final RegularFileProperty getTsConfig() {
     @TaskAction
     public final void generateTsConfig() throws IOException {
         Set<File> externalDependencyFiles = getCompileClasspath()
-                .filter(element -> !element.getAbsolutePath()
-                        .startsWith(
-                                getProject().getRootProject().getProjectDir().getAbsolutePath()))
+                .filter(element -> !isProjectDependency(element))
                 .getFiles();
 
-        Map<String, List<Path>> projectDepsFiles = getCompileClasspath()
-                .filter(element -> element.getAbsolutePath()
-                        .startsWith(
-                                getProject().getRootProject().getProjectDir().getAbsolutePath()))
-                .getFiles()
-                .stream()
+        Map<String, List<Path>> projectDepsFiles = getCompileClasspath().getFiles().stream()
+                .filter(this::isProjectDependency)
```

Comment:
we should either use gradle or java's api to filter

Reply:
I just did this for simplicity to avoid another collect